src/API/Site/Controllers/Assets/AssetSuggestionsController.php

```diff
@@ -0,0 +1,162 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Assets;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AssetSuggestionsService;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use WP_REST_Request as Request;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class AssetSuggestionsController
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Assets
+ */
+class AssetSuggestionsController extends BaseController {
+
+	/**
+	 * Service used to populate ads suggestions data.
+	 *
+	 * @var AssetSuggestionsService
+	 */
+	protected $asset_suggestions_service;
+
+	/**
+	 * AssetSuggestionsController constructor.
+	 *
+	 * @param RESTServer              $server
+	 * @param AssetSuggestionsService $asset_suggestion
+	 */
+	public function __construct( RESTServer $server, AssetSuggestionsService $asset_suggestion ) {
+		parent::__construct( $server );
+		$this->asset_suggestions_service = $asset_suggestion;
+	}
+
+	/**
+	 * Register rest routes with WordPress.
+	 */
+	public function register_routes(): void {
+		$this->register_route(
+			'assets/final-url/suggestions',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_final_urls_suggestions_callback(),
```

Comment:
💅  minor

`get_final_urls_suggestions_callback `has `final_urls` (plural) in the name whereas the endpoint is in singular `final-url`